SPEC.md

```diff
@@ -201,25 +201,161 @@ Note that logging mechanisms other than stdout and stderr are not required by th
+The first example is "capabilities" but this will be expanded to include things such as SELinux, SMACK or AppArmor.
+
+#### linux/capabilities-remove-set
+
+* Scope: app
+
+**Parameters:**
+* **set** list of capabilities that will be removed from the process's capabilities bounding set
+
+```
+"name": "linux/capabilities-remove-set",
+"set": [
+	"CAP_SYS_PTRACE",
+]
+```
+
+#### linux/capabilities-retain-set
+
+* Scope: app
+
+**Parameters:**
+* **set** list of capabilities that will be retained in the process's capabilities bounding set
+
+```
+"name": "linux/capabilities-retain-set",
+"set": [
+	"CAP_KILL",
+	"CAP_CHOWN"
+]
+```
+
+### Resource Isolators
+
+#### resource/cpu
+
+* Scope: app/container
+
+**Parameters:**
+* **request** milli-cores that are requested
+* **limit** milli-cores that can be consumed before the Kernel temporarily throttles the process
+
+```
+"name": "resource/cpu",
+"request": "250",
+"limit": "500"
+```
+
+**Note**: a milli-core is the milli-seconds/second that the app will be able to run. e.g. 1000 would represent full use of a single CPU core every second.
+
+#### resource/memory
+
+* Scope: app/container
+
+**Parameters:**
+* **request** bytes of memory that the app is requesting to use and will be reclaimed to in case of contention
+* **limit** bytes of memory that the app can allocate before the kernel considers the container out of memory and stops allowing allocations.
+
+```
+"name": "resource/memory",
+"request": "1G",
+"limit": "2G"
+```
+
+#### resource/block-weight
+
+* Scope: app/container
+
+**Parameters:**
+* **default** if set to true this weight will be the default for all devices. If set devicePaths must be empty.
+* **devicePaths** the block devices that this weight will be placed on.
+* **request** read/write fraction of device I/O time between 1 and 100 allocated to this container
+
+```
+"name": "resource/block-weight",
+"default": false,
+"devicePaths": [
+	"/dev/sda"
+],
+"request": {
+	"read": "100",
+	"write": "100"
+}
+```
+
+#### resource/block-bandwidth
+
+* Scope: app/container
+
+**Parameters:**
+* **devicePaths** the block devices that this limit will be placed on
+* **limit** read/write bytes per second
+
+```
+"name": "resouce/block-bandwidth",
+"devicePaths": [
+	"/dev/sda"
+],
+"limit": {
+	"read": "2MB",
+	"write": "512MB"
+}
+```
+
+#### resource/block-iops
+
+* Scope: app/container
+
+**Parameters:**
+* **devicePaths** the block devices that this limit will be placed on
+* **limit** read/write input/output operations per second
+
+```
+"name": "resource/block-iops",
+"devicePaths": [
+	"/dev/sda",
+],
+"limit": {
+	"read": "2000",
+	"write": "512"
+}
+```
+
+#### resource/network-bandwidth
+
+* Scope: app/container
+
+**Parameters:**
+* **default** must be set to true and means that this bandwidth limit applies to all interfaces except localhost by default.
+* **limit** read/write bytes per second
```

Comment:
I think it should work but I don't know if it'll always be possible to police on the host side. Depending on the type of network interface, you may need to police on the container side. If done on the container side, an app can just remove the limits (if it has proper capabilities).